spec/cspec/c/kernel.mk

```diff
@@ -8,29 +8,39 @@
 # It allows building the C kernel in locations other than the default one used by l4v,
 # and assumes that KERNEL_BUILD_ROOT has already been set to specify the build location.
 
+ifndef KERNEL_BUILD_ROOT
+  $(error KERNEL_BUILD_ROOT is not set)
+endif
+
 ifndef L4V_REPO_PATH
   L4V_REPO_PATH := $(realpath $(dir $(lastword ${MAKEFILE_LIST}))../../..)
 endif
 
-ifndef SOURCE_ROOT
-  SOURCE_ROOT := $(realpath ${L4V_REPO_PATH}/../seL4)
+ifndef KERNEL_SOURCE_ROOT
+  KERNEL_SOURCE_ROOT := $(realpath ${L4V_REPO_PATH}/../seL4)
 endif
 
 CSPEC_DIR := ${L4V_REPO_PATH}/spec/cspec
-PARSERPATH := ${L4V_REPO_PATH}/tools/c-parser/standalone-parser
 
 ifndef L4V_ARCH
   $(error L4V_ARCH is not set)
 endif
 
+SEL4_CONFIG_NAME := ${L4V_ARCH}$(if ${L4V_FEATURES},_${L4V_FEATURES},)
+
 ifndef CONFIG
-  CONFIG := ${SOURCE_ROOT}/configs/${L4V_ARCH}_$(if ${L4V_FEATURES},${L4V_FEATURES}_,)verified.cmake
+  CONFIG := ${KERNEL_SOURCE_ROOT}/configs/${SEL4_CONFIG_NAME}_verified.cmake
 endif
 
 ifndef CONFIG_DOMAIN_SCHEDULE
   CONFIG_DOMAIN_SCHEDULE := ${CSPEC_DIR}/c/config_sched.c
 endif
 
+# Default to the setting in the CONFIG file.
+ifdef CONFIG_OPTIMISATION
+  KERNEL_CMAKE_OPTIMISATION := -DKernelOptimisation=${CONFIG_OPTIMISATION}
+endif
```

Comment:
which `CONFIG` file is the comment referring to?

Reply:
The `CONFIG` variable a few lines up. I'll make that more explicit, by changing it to `${CONFIG}`.